### DISCUSSION TITLE
[Help]: Accessing options from API

### DISCUSSION THREAD

**Discussion by skinread** [davidjerleke/embla-carousel#1035] [Help] [answered]
### Summary

Using Embla with React via shadcn/ui, I have found that I need to update some custom states based on the current breakpoint. Configuring breakpoints with `options` is easy. However I can't find an api method determine which breakpoint behaviour is currently applied so I can update custom state responsively.

Example implementation below for custom state that works easily based on the existing api methods that are updated on init/reInit/select events. I believe that each time Embla changes config using `breakpoints` option, the reInit event is fired.

Is there a way to check current `options` from the api? Would it be through the `internalEngine`?

What I'd like to be able to do is retrieve something like a the current option set (at the current breakpoint), or the current viewport size that Embla has observed. Any of these would help me.

Example of what I'd like to do inside the useCallback:
Assuming `options` set with breakpoints
```js
options={{ 
  loop: true,
  breakpoints: { '(min-width: 768px)': { loop: false }}
}
```
then
```jsx
const onUpdate = useCallback((api) => {
  if (!api) return;

  // would set the state based on value of a particular options, and change responsively
  setMyCustomState(api.currentOptions().loop);
}, []);
```

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-auto-height

### Additional information

Example of custom state from the api that is working:

```js
const [canScrollPrev, setCanScrollPrev] = useState(false);
const [canScrollNext, setCanScrollNext] = useState(false);

const onUpdate = useCallback((api) => {
  if (!api) {
    return;
  }

  setCanScrollPrev(api.canScrollPrev());
  setCanScrollNext(api.canScrollNext());
}, []);

const scrollPrev = useCallback(() => {
  emblaApi?.scrollPrev();
}, [emblaApi]);

const scrollNext = useCallback(() => {
  emblaApi?.scrollNext();
}, [emblaApi]);

useEffect(() => {
  if (!emblaApi || !setApi) {
    return;
  }

  setApi(emblaApi);
}, [emblaApi, setApi]);

useEffect(() => {
  if (!emblaApi) {
    return;
  }

  onUpdate(emblaApi);
  emblaApi.on('init', onUpdate);
  emblaApi.on('reInit', onUpdate);
  emblaApi.on('select', onUpdate);

  return () => {
    emblaApi?.off('select', onUpdate);
  };
}, [emblaApi, onUpdate]);
```

**Comment by sarussss** [comment, marked as answer]
Hi @skinread You can try 

```tsx
emblaApi.internalEngine().options
emblaApi.internalEngine().options.loop
```

**Reply by skinread** [reply]
Thank you, so obvious now.
